### PR TITLE
Add upgrade guide for Sensu Go

### LINKED ISSUE #845: Sensu Go upgrade guide
Creating a new page at /install/upgrade that includes:

- Outline of difference between 1.x and 5.0
- How to upgrade from 1.x to 5.0: Link to blog post

Notes:

- https://github.com/sensu/sensu-translator

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

First draft for upgrade guide.

To do:

- Resolve `...`
- Add time window callouts

## Motivation and Context

Closes #845

### CODE REVIEW

File: content/sensu-go/5.0/installation/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+title: "Upgrading to Sensu Go"
+linkTitle: "Upgrade Sensu"
+description: "Guide to upgrading to Sensu Go from Sensu Core"
```
Comment (bananabele) [inline content/sensu-go/5.0/installation/upgrade.md]: I'd suggest changing "Sensu Core" to "Sensu 1.x"

File: content/sensu-go/5.0/installation/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+- [Assets](#assets)
+- [Role-based access control](#role-based-access-control)
+- [Silencing](#silencing)
+- [Token substitution](#token-substitution)
+- [Aggregates](#aggregates)
+- [API](#api)
+
+### Packaging
+Sensu is now provided as three packages: sensu-backend, sensu-agent, and sensu-cli (sensuctl).
+This results in a fundamental change in Sensu terminology from Sensu Core: the server is now the backend; the client is now the agent.
```
Comment (bananabele) [inline content/sensu-go/5.0/installation/upgrade.md]: per my comment above, sub "Sensu Core" for "Sensu Core 1.6"

File: content/sensu-go/5.0/installation/upgrade.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+To learn more about new terminology in Sensu Go, see the [glossary][1].
+
+### Architecture
+The external RabbitMQ transport and Redis datastore in Sensu Core have been replaced with an embedded transport and [etcd datastore][2] in Sensu Go.
```
Comment (bananabele) [inline content/sensu-go/5.0/installation/upgrade.md]: "Sensu Core 1.6"

File: content/sensu-go/5.0/installation/upgrade.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+### Checks
+Standalone checks are no longer supported in Sensu Go, although similar functionality can be achieved using role-based access control, assets, and entity subscriptions.
+There are also a few changes to check definitions to be aware of. The `stdin` check attribute is no longer supported in Sensu Go, and Sensu Go no longer tries to run a "default" handler when executing a check without a specified handler. Additionally, round-robin subscriptions are not yet available in Sensu Go 5.0
+
+[Check hooks][8] are now a resource type in Sensu Go, meaning that hooks can be created, managed, and reused independently of check definitions.
+You can also execute multiple hooks for any given response code.
+
+### Events
+All check results are now considered events and are processed by event handlers.
+You can use the built-in [incidents filter][9] to recreate the Sensu Core behavior in which only non-zero check results are considered events.
```
Comment (bananabele) [inline content/sensu-go/5.0/installation/upgrade.md]: "Sensu Core 1.6"

File: content/sensu-go/5.0/installation/upgrade.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+### Handlers
+Transport handlers are no longer supported by Sensu Go, but you can create similar functionality using a pipe handler that connects to a message bus and injects event data into a queue.
+
+### Filters
+Ruby eval logic has been replaced with JavaScript expressions in Sensu Go, opening up powerful possibilities to filter events based on occurrences and other event attributes.
+As a result, the built-in occurrences filter in Sensu Core is not provided in Sensu Go, but you can replicate its functionality using [this filter definition][10].
```
Comment (bananabele) [inline content/sensu-go/5.0/installation/upgrade.md]: "Sensu Core 1.6"

File: content/sensu-go/5.0/installation/upgrade.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+Sensu Go includes three [new built-in filters][11]: only-incidents, only-metrics, and allow-silencing.
+Sensu Go 5.0 does not yet include the built-in check dependencies filter from Sensu Core.
```
Comment (bananabele) [inline content/sensu-go/5.0/installation/upgrade.md]: "Sensu Core 1.6"

File: content/sensu-go/5.0/installation/upgrade.md (hunk 1 above)
Comment (Blue0ctober) [reply content/sensu-go/5.0/installation/upgrade.md]: We may want to specify a specific version such as 1.3 and above. I believe we do have a cut-off for support?

File: content/sensu-go/5.0/installation/upgrade.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+To learn more about setting up RBAC in Sensu Go, see the [RBAC reference][13] and the [guide to creating a read-only user][14].
+
+### Silencing
+Silencing is now disabled by default in Sensu Go and must be enabled explicitly using the built-in [`not_silenced` filter][15].
+
+### Token substitution
+The syntax for using token substitution has changed from using triple colons to using [double curly braces][16].
+
+### Aggregates
+Sensu Go 5.0 does not yet support aggregates.
```
Comment (apaskulin) [inline content/sensu-go/5.0/installation/upgrade.md]: ...support check aggregates...

File: content/sensu-go/5.0/installation/upgrade.md (hunk 4 above)
Comment (cwjohnston) [inline content/sensu-go/5.0/installation/upgrade.md]: how about rewording this sentence like so:

"You can use the built-in incidents filter to recreate the Sensu Core behavior in which only check results with a non-zero status are considered events."

**Comment by jspaleta** [comment]
I think its reasonable for GA. Hopefully we'll update this in the next week once the new sensu-plugin gem is released.
